File: backend/src/main/java/com/apocalypse/thefall/controller/skill/SkillController.java
package com.apocalypse.thefall.controller.skill;

import com.apocalypse.thefall.dto.character.skill.SkillDto;
import com.apocalypse.thefall.mapper.SkillMapper;
import com.apocalypse.thefall.service.skill.SkillService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/api/skills")
@RequiredArgsConstructor
public class SkillController {

    private final SkillService skillService;
    private final SkillMapper skillMapper;

    @GetMapping("/all")
    public List<SkillDto> getAll() {
        return skillMapper.toDto(skillService.findAll());
    }
}

File: backend/src/main/java/com/apocalypse/thefall/controller/character/stats/SkillController.kt
package com.apocalypse.thefall.controller.character.stats

import com.apocalypse.thefall.dto.character.stats.DataItemDto
import com.apocalypse.thefall.mapper.character.stats.skill.toListDataItemDto
import com.apocalypse.thefall.service.character.stats.SkillService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/skills")
class SkillController(
    private val skillService: SkillService
) {

    @GetMapping("/all")
    fun getAll(): List<DataItemDto> = skillService.findAll().toListDataItemDto()
}